src/test/java/io/thinkit/edc/client/connector/ContractDefinitionsTest.java

```diff
@@ -39,6 +41,27 @@ void should_get_a_contract_definition() {
         assertThat(contractDefinition.getContent().createdAt()).isGreaterThan(0);
     }
 
+    @Test
+    void should_get_a_contract_definition_async() {
+        try {
+            Result<ContractDefinition> contractDefinition =
+                    contractDefinitions.getAsync("definition-id").get();
+
+            assertThat(contractDefinition.isSucceeded()).isTrue();
```

Comment:
assertions on the result are equal to the ones in the "sync" test, they could be refactored as well